src/pipeline.js

```diff
@@ -16,16 +16,48 @@ export default class Pipeline {
   run() {
     this.listener.listen((body, _headers) => {
       this.logger.debug(`received message: ${body}`)
-      let uri = JSON.parse(body).object.id
-      this.logger.debug(`uri needs indexing: ${uri}`)
-      new Request(uri).body()
-        .then(json => {
-          this.logger.debug(`indexing resource: ${JSON.stringify(json)}`)
-          this.indexer.index(json)
-        })
-        .catch(err => {
-          this.logger.error(`error processing ${uri}: ${err.message}`)
-        })
+      body = JSON.parse(body)
+
+      let uri = body.object.id
+      this.logger.debug(`resource ${uri} needs indexing`)
+
+      // `body.type` looks like { ... "type": [ "http://www.w3.org/ns/prov#Activity", "Delete" ] ... }
+      let operation = body.type[1].toLowerCase()
+      switch(operation) {
+        case 'update':
+        case 'create':
+        case 'delete':
+          // This works because the expressions in the above case statement are
+          // functions in this class
+          this[operation](uri)
+          break
+        // We only understand the above operation types. Log an error and keep listening.
+        default:
+          this.logger.error(`unsupported operation: ${operation}`)
+      }
     })
   }
+
+  // TODO: Figure out a more Javascript-y way to have class functions share behavior
```

Comment:
Maybe using a JS functional approach?

Reply:
Yeah, I just don't know what that means at this point. :laughing: All of my refactoring experience and knowledge is OOP-specific, other than DRY (blech) and extracting small functions from larger functions.